crates/volta-core/src/run_package_global/binary.rs

```diff
@@ -0,0 +1,165 @@
+        exe,
+        args,
+        None,
+        ToolKind::DefaultBinary(bin),
+    ))
+}
+
+/// Determine the execution context (PATH and failure error message) for a project-local binary
+pub(super) fn local_execution_context(
+    tool: String,
+    platform: Option<Platform>,
+    session: &mut Session,
+) -> Fallible<(OsString, ErrorKind)> {
+    match platform {
+        Some(plat) => {
+            let image = plat.checkout(session)?;
+            let path = image.path()?;
+            debug_active_image(&image);
+
+            Ok((
+                path,
+                ErrorKind::ProjectLocalBinaryExecError { command: tool },
+            ))
+        }
+        None => {
+            let path = System::path()?;
+            debug_no_platform();
+
+            Ok((path, ErrorKind::NoPlatform))
+        }
+    }
+}
+
+/// Determine the execution context (PATH and failure error message) for a default binary
+pub(super) fn default_execution_context(
+    tool: String,
+    platform: Option<Platform>,
+    session: &mut Session,
+) -> Fallible<(OsString, ErrorKind)> {
+    match platform {
+        Some(plat) => {
+            let image = plat.checkout(session)?;
+            let path = image.path()?;
+            debug_active_image(&image);
+
+            Ok((path, ErrorKind::BinaryExecError))
+        }
+        None => {
+            let path = System::path()?;
+            debug_no_platform();
+
+            Ok((path, ErrorKind::BinaryNotFound { name: tool }))
+        }
+    }
+}
+
+/// Information about the location and execution context of default binaries
+///
+/// Fetched from the config files in the Volta directory, represents the binary that is executed
+/// when the user is outside of a project that has the given bin as a dependency.
+pub struct DefaultBinary {
+    pub bin_path: PathBuf,
+    pub platform: Platform,
+}
+
+impl DefaultBinary {
+    pub fn from_config(bin_config: BinConfig, session: &mut Session) -> Fallible<Self> {
+        let package_dir = volta_home()?.package_image_dir(&bin_config.package);
+        let mut bin_path = bin_config.manager.binary_dir(package_dir);
+        bin_path.push(&bin_config.name);
+
+        // If the user does not have yarn set in the platform for this binary, use the default
+        // This is necessary because some tools (e.g. ember-cli with the `--yarn` option) invoke `yarn`
+        let yarn = match bin_config.platform.yarn {
+            Some(yarn) => Some(yarn),
+            None => session
+                .default_platform()?
+                .and_then(|ref plat| plat.yarn.clone()),
+        };
+        let platform = Platform {
+            node: Sourced::with_binary(bin_config.platform.node),
+            npm: bin_config.platform.npm.map(Sourced::with_binary),
+            yarn: yarn.map(Sourced::with_binary),
+        };
+
+        Ok(DefaultBinary { bin_path, platform })
+    }
+
+    pub fn from_name(tool_name: &OsStr, session: &mut Session) -> Fallible<Option<Self>> {
```

Comment:
I think we should have *either* a doc-string explaining what `Some` and `None` mean, or make it a custom `enum` indicating `Found { bin_path, platform }` – probably the former! Just that way it's clear to any future readers (including us in a year 😂) what the intent is.

Reply:
Done!